src/graph.py

```diff
@@ -1,82 +1,109 @@
 
+
 class GNode(object):
     """Create a GNode Constructor."""
 
     def __init__(self, value):
         """Create an instance of a GNode with the attribues of a value and a list of edges."""
         self.value = value
-        self.edges = []
+        self.edges = set()
+
+
+class Edge(object):
+    """Create an Edge Constructor."""
+
+    def __init__(self, begin, end):
+        """Create an instance of an edge."""
+        self.begin = begin
+        self.end = end
+
+    def __eq__(self, other):
+        """Allow comparison of two equal edges."""
+        return self.begin == other.begin and self.end == other.end
+
+    def __ne__(self, other):
+        """Allow comparison of two unequal edges."""
+        return self.begin != other.begin or self.end != other.end
 
 
 class Graph(object):
     """Create a Graph Constructor."""
 
     def __init__(self):
         """Create an instance of a Graph that has a list of nodes."""
-        self.list_of_nodes = []
+        self.set_of_nodes = set()
 
     def add_node(self, new_node):
-        """Adds a new node to the graph"""
-        self.list_of_nodes.append(new_node)
+        """Add a new node to the graph."""
+        self.is_node(new_node)
+        self.set_of_nodes.add(new_node)
 
+    def is_node(self, node):
+        """Check to see if object is a node. Returns a TypeError if it is not a node."""
+        if isinstance(node, GNode):
+            return True
+        else:
+            raise TypeError('{} is not a GNode type.'.format(str(node)))
 
     def has_node(self, node):
```

Comment:
Consider this: how and under what circumstances would a user have access to a specific node in order to check whether it is in the Graph? There would have to be a whole separate process of creating GNodes outside of the scope of the Graph, duplicating a lot of effort and making the Graph almost redundant.